<a href="https://colab.research.google.com/github/pgallino/Organizacion-de-datos-TP1-1C2023/blob/main/pandas54.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


54) ¿Cuál es la acción más realizada por usuarios que no están registrados? (⭐⭐)

In [ ]:
df_logs = pd.read_csv('/content/drive/MyDrive/Wikipedia dataset 2c2021/logs.csv')
df_logs.head(20)

<ipython-input-3-a2efcc830d68>:1: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  df_logs = pd.read_csv('/content/drive/MyDrive/Wikipedia dataset 2c2021/logs.csv')


item_id             timestamp contributor_username  contributor_id  \
0         1  2004-12-23T07:10:09Z               Sabbut          2075.0   
1         2  2004-12-23T07:14:09Z               Sabbut          2075.0   
2         3  2004-12-23T07:26:10Z               Sabbut          2075.0   
3         4  2004-12-23T07:55:09Z               Sanbec          7856.0   
4         5  2004-12-23T08:02:22Z               Sanbec          7856.0   
5         6  2004-12-23T08:28:32Z                 Dodo         13770.0   
6         7  2004-12-23T12:10:06Z              Antoine         22532.0   
7         8  2004-12-23T12:12:15Z              Antoine         22532.0   
8         9  2004-12-23T12:13:43Z              Antoine         22532.0   
9        10  2004-12-23T12:23:54Z               Sanbec          7856.0   
10       11  2004-12-23T12:56:24Z               Sanbec          7856.0   
11       12  2004-12-23T13:54:16Z               Sanbec          7856.0   
12       13  2004-12-23T14:39:43Z        Gabriel Abril         18805.0   
13       14  2004-12-23T14:43:41Z              JorgeGG          2906.0   
14       15  2004-12-23T15:19:52Z             Ascánder         14862.0   
15       16  2004-12-23T15:25:36Z              JorgeGG          2906.0   
16       17  2004-12-23T15:29:01Z             Ascánder         14862.0   
17       18  2004-12-23T15:29:22Z             Ascánder         14862.0   
18       19  2004-12-23T15:31:17Z             Ascánder         14862.0   
19       20  2004-12-23T15:31:39Z             Ascánder         14862.0   

   contributor_ip                                            comment logtype  \
0             NaN  el contenido era: 'Si aprendés Pensamiento Cie...  delete   
1             NaN  el contenido era: '{{destruir}}en algun lugar ...  delete   
2             NaN            el contenido era: '#REDIRECT [[Flash]]'  delete   
3             NaN  Sin información de origen ni de licencia. Hay ...  delete   
4             NaN  Sustituida por  [[commons:Image:Heckert_GNU.png]]  delete   
5             NaN                                 vandalismo grosero   block   
6             NaN  Logotipo Comuna de La Florida, derecho público...  upload   
7             NaN  Fotografía aérea instantánea de los videos de ...  upload   
8             NaN      Pablo Zalaquett, Alcalde de La Florida {{PD}}  upload   
9             NaN                                    Está en commons  delete   
10            NaN  el contenido era: ''''Pueblo''' de 1200 habita...  delete   
11            NaN  el contenido era: '#REDIRECT [[Ayerbe (Huesca)]]'  delete   
12            NaN                                  tags pasivos RFID  upload   
13            NaN  reparando un aparente problema de la base de d...  delete   
14            NaN                                            Copyvio  delete   
15            NaN  el contenido era: '{{destruir}}' anteriormente...  delete   
16            NaN                   el contenido era: '{{destruir}}'  delete   
17            NaN                   el contenido era: '{{destruir}}'  delete   
18            NaN            el contenido era: '* [[Daniel Pennac]]'  delete   
19            NaN                   el contenido era: '{{destruir}}'  delete   

    action                                       title  
0   delete                            Verificacionismo  
1   delete                                    Rapunzel  
2   delete                            Macromedia Flash  
3   delete                        Archivo:Gnu head.png  
4   delete                     Archivo:Gnu-head-sm.jpg  
5    block                       Usuario:212.55.30.157  
6   upload                      Archivo:La florida.jpg  
7   upload                 Archivo:La florida uoct.jpg  
8   upload                 Archivo:Pablo zalaquett.jpg  
9   delete              Archivo:Szymborska znaczek.jpg  
10  delete                           Carabaña (Madrid)  
11  delete                                      AYERBE  
12  upload                 

filtro el dataframe por los usuarios sin id ya que son los que no tienen usuario

In [ ]:
df_nr = df_logs.loc[df_logs['contributor_id'].isnull()]

Aparece un contributor_username 'Unknown user' pero es similar a NaN, no es un usuario real.

In [ ]:
df_nr

item_id             timestamp contributor_username  contributor_id  \
21394        21395  2005-05-27T16:31:25Z         Unknown user             NaN   
42406        42407  2005-08-28T05:11:12Z                  NaN             NaN   
42416        42417  2005-08-28T08:20:23Z                  NaN             NaN   
57522        57523  2005-09-30T01:23:43Z                  NaN             NaN   
79235        79236  2005-11-05T02:19:13Z                  NaN             NaN   
...            ...                   ...                  ...             ...   
12188877  15587331  2021-09-03T02:14:16Z                  NaN             NaN   
12188881  15587335  2021-09-03T02:18:51Z                  NaN             NaN   
12188913  15587367  2021-09-03T02:54:49Z                  NaN             NaN   
12188915  15587369  2021-09-03T02:56:15Z                  NaN             NaN   
12188918  15587372  2021-09-03T02:57:44Z                  NaN             NaN   

                                 contributor_ip  \
21394                                       NaN   
42406                                       NaN   
42416                                       NaN   
57522                                       NaN   
79235                                       NaN   
...                                         ...   
12188877                         211.251.53.228   
12188881  2806:261:400:16C4:A52F:D057:7E1C:499A   
12188913                         190.122.184.86   
12188915    2806:108E:1C:366A:EC17:C8DB:35:3EAF   
12188918                         186.73.165.140   

                                                    comment logtype   action  \
21394                                                   NaN  upload   upload   
42406                                                   NaN   block    block   
42416                                                   NaN   block  unblock   
57522                                  Mass cross-wiki spam  delete   delete   
79235                                                   NaN    move     move   
...                                                     ...     ...      ...   
12188877                  Página redirigida a [[GLOBAL 24]]  create   create   
12188881  Sección nueva: /* robo de espiritualidad mente...  create   create   
12188913                                              Cooll  create   create   
12188915  Página creada con «{{Ficha de libro |nombre   ...  create   create   
12188918  Página creada con «Elemento elaborado en mader...  create   create   

                                          title  
21394                           Archivo:123.jpg  
42406                                       NaN  
42416                                       NaN  
57522                               W/index.php  
79235                                       NaN  
...                                         ...  
12188877                              Global 24  
12188881  Discusión:Defensa nacional espiritual  
12188913                        Chiquimula F.C.  
12188915                Black Hawk Down (libro)  
12188918                   Bastón Sonajero Guna  

[125253 rows x 9 columns]

Agrupo por action, para contar la cantidad de veces que fue realizada cada una

In [ ]:
df_mas_realizados = df_nr.groupby(['action']).agg({'item_id':'count'})

In [ ]:
df_mas_realizados.columns = ['veces_realizada']
df_mas_realizados

veces_realizada
action                     
autocreate              239
block                    50
byemail                   2
create               124728
create2                  16
delete                  101
event                    17
move                     10
newusers                  4
protect                  22
reblock                   1
renameuser                3
restore                   3
revision                 36
thank                     1
unblock                  13
update                    6
upload                    1

In [ ]:
maxima_cantidad = df_mas_realizados['veces_realizada'].max()
df_mas_realizada = df_mas_realizados.loc[df_mas_realizados['veces_realizada'] >= maxima_cantidad]
df_mas_realizada

veces_realizada
action                 
create           124728

Por lo que podemos ver que el máximo es create con 124728